src/libmongoc/tests/test-mongoc-collection.c

```diff
@@ -4016,7 +4016,7 @@ _test_insert_validate (insert_fn_t insert_fn)
 }
 
 static void
-test_insert_bulk_validate (void)
+test_insert_bulk_validate (void *ctx)
```

Comment:
These tests use [error API 2](http://mongoc.org/libmongoc/current/errors.html#setting-the-error-api-version), which means all server-side errors should be reported in the `MONGOC_ERROR_SERVER` domain. The expectations below use `MONGOC_ERROR_COMMAND` as the domain, which tells me these are client-side errors.

In that case, why should the server version have any impact on these tests? If anything, I would expect these tests to be removed entirely as part of [CDRIVER-3895](https://jira.mongodb.org/browse/CDRIVER-3895) since that will remove all client-side validation of dots and dollars. Alternatively, we could change the documents in these tests to violate some other validation rule that will be left in place (e.g. invalid UTF).

Reply:
This test fails with an assertion error: https://evergreen.mongodb.com/task_log_raw/mongo_c_driver_windows_2017_test_latest_sharded_noauth_nosasl_nossl_patch_d08ef111e1a9cf7e71d916f562b46a9d9bf061c6_60a625487742ae6f40858e18_21_05_20_09_01_04/0?type=T#L2378. You are correct that the first bulk insert expects a command error because it uses no insert validation. There is a second insert that specifies `MONGOC_INSERT_NO_VALIDATE` which expectedly fails on 5.0 and expects a server error.

I was considering rewriting the tests but in the interest of fixing the build so I can get on with the language refactoring (which has the potential of breaking lots of things) I decided to skip these tests for now and revisit them when we implement changes for handling field names with dots and dollars. I've added a TODO to re-enable the tests (also fixed clang-format while I was at it)